### DISCUSSION TITLE
About SplineMesh

### DISCUSSION THREAD

**Discussion by Makunugi** [cables-gl/cables_docs#294] [Answers & Questions] [answered]
hello.
How can I create a spline that follows the position of  the object in 3D space on one side and sticks to the screen on the other side, like the red line in the video?
I couldn't make a sample with Cables, so I made a video with After Effects.

https://user-images.githubusercontent.com/104124223/178476739-7328b739-ab94-4497-807a-0780991f472f.mp4

**Comment by tastez** [comment, marked as answer]
nice question! It becomes tricky but doable if you need your text to be in HTML. If you don't then it is a lot easier. I'll prepare a solution, let me know if you want HTML or not

**Reply by Makunugi** [reply]
thank you!
No HTML required.

**Reply by tastez** [reply]
I usually use https://cables.gl/op/Ops.Gl.Matrix.GetModelMatrix with https://cables.gl/op/Ops.Array.Array3GetNumbers and set Array3GetNumber to index=4 , this way i get the location of the object that i connect GetModelMatrix to
![image](https://user-images.githubusercontent.com/78642112/178492684-30a9917b-1fb2-4740-89bb-0898885a1c12.png)

then you need to do the same for your TextMesh
![image](https://user-images.githubusercontent.com/78642112/178492796-aaa1930c-9196-41bd-9fe1-b7632cd01c4b.png)

and now you have 2 XYZ points that you can use to create an Array for the Spline

maybe there is a better way. Anyway here is what it can look like
https://cables.gl/p/62cd655c9463228768b12cc5?s=foaklRaFnAWBXl3d

**Reply by Makunugi** [reply]
thank you! it's great! I wanted to do this!
I learned a lot!

**Reply by tastez** [reply]
you can also add a Transform under the 3d model and then the GetModelMatrix after that..so you can easily adjust location of the spline point.
![image](https://user-images.githubusercontent.com/78642112/178496824-fa166931-995e-4595-8ff2-8469f8a7ff16.png)


and for making the spline points array im lazy so just using ArrayFromNumbers
![image](https://user-images.githubusercontent.com/78642112/178496957-1503da36-6cd5-4cde-b7c6-451d075c0816.png)

**Reply by Makunugi** [reply]
I was thinking of doing that too.
Thank you for letting me know in detail!
It's even more fun!
